docs/orchestrating-elastic-stack-applications/logstash.asciidoc

```diff
@@ -874,12 +874,146 @@ spec:
+
+This example installs the same plugin, but uses a custom image to install the plugin.
+
+First, create a Dockerfile based on the {ls} Docker image, running `bin/logstash-plugin install` to install the appropriate plugin(s):
+
+[subs="attributes,+macros,callouts"]
+----
+FROM docker.elastic.co/logstash/logstash:{version}
+
+RUN bin/logstash-plugin install logstash-filter-tld
+----
+
+Then after building and deploying the custom image (refer to <<{p}-custom-images>> for more details), include it in the {ls} manifest:
+
+[source,yaml]
+----
+spec:
+  count: 1
+  version: {version} <1>
+  image: <CUSTOM_IMAGE>
+----
+<1> Providing the correct version is always required as ECK reasons about APIs and capabilities available to it based on the version field.
+
+
+[id="({p}-logstash-jar-files"]
+== Working with plugins that require additional files
+
+Running {ls} may require additional files, such as JAR files needed to load JDBC drivers when using a JDBC or JMS plugin. To add these files, there are two options available - using an `initContainer` to add files before the main container start, or creating a custom Docker image that includes the required files. Refer to <<{p}-bundles-plugins>> for a run down of which option might be most suitable for you.
+
+
+=== Adding files using an initContainer
+
+This example creates an `initContainer` to download a PostgreSQL JDBC driver JAR file, and place it in a volume mount accessible to the main `container`, and then use it in a JDBC input in the pipeline configuration.
+
+[source,yaml]
+----
+spec:
+  podTemplate:
+    spec:
+      initContainers:
+      - name: download-postgres
+        command: ["/bin/sh"]
+        args: ["-c", "curl -o /data/postgresql.jar -L https://jdbc.postgresql.org/download/postgresql-42.6.0.jar"]
+        volumeMounts:
+          - name: workdir
+            mountPath: /data
+      containers:
+        - name: logstash
+          volumeMounts:
+            - name: workdir
+              mountPath: /usr/share/logstash/jars <1>
+  pipelines:
+    - pipeline.id: main
+      config.string: |
+        input {
+          jdbc {
+             jdbc_driver_library => "/usr/share/logstash/jars/postgresql.jar" <1>
+             jdbc_driver_class => "org.postgresql.Driver"
+                :
+                :
+          }
+        }
+----
+<1> Referring to the external file should match the `mountPath` of the `container`
+
+=== Adding files using a custom image
+
+This example downloads the same `postgres` JDBC driver, and adds it to the {ls} classpath in the Docker image.
+
+First, create a Dockerfile based on the {ls} Docker image, and download the JDBC driver, storing it alongside the other JAR files in the {ls} classpath:
+
+[source,subs="attributes,+macros,callouts"]
+----
+FROM docker.elastic.co/logstash/logstash:{version}
+
+RUN curl -o /usr/share/logstash/logstash-core/lib/jars/postgresql.jar -L https://jdbc.postgresql.org/download/postgresql-42.6.0.jar <1>
```

Comment:
This is not working as expected.  The URL is interpreted as a hyperlink (href).  

<img width="709" alt="Screen Shot 2024-01-17 at 4 47 38 PM" src="https://github.com/elastic/cloud-on-k8s/assets/35154725/c6ab3ed6-efa1-44ff-af47-e5a34c23a8f5">

I can look for examples and/or play around with how to fix this.

Reply:
I have it looking right with this `["source","sh",subs="attributes,callouts"]`, but CURL isn't working: 

<img width="708" alt="Screen Shot 2024-01-17 at 5 08 28 PM" src="https://github.com/elastic/cloud-on-k8s/assets/35154725/8e084bad-3712-446a-a1d2-d4ef510c1d8a">

I'll keep looking.